iis-wf/iis-wf-primary/src/test/resources/eu/dnetlib/iis/wf/primary/processing/sampledataproducer/oozie_app/workflow.xml

```diff
@@ -482,6 +482,22 @@
                     <name>patentServiceEndpointOpsUriRoot</name>
                     <value>expectedOpsUriRoot</value>
                 </property>
+				<property>
+					<name>affiliation_matching_number_of_emitted_files</name>
+					<value>1</value>
+				</property>
+				<property>
+					<name>documentsclassification_main_number_of_partitions</name>
```

Comment:
Could we align this property name with all the other cases by renaming:

`documentsclassification_main_number_of_partitions`

into:

`documentsclassification_number_of_emitted_files`

so I would skip the unneccessary `_main` part and rename `partitions` to `emitted_files`.

Reply:
I've just realized that the original change may be a bit far fetched. The parameter does not directly correspond to a repartition at data store write. The repartition happens a few transformations earlier and is followed by maps, filters and a pipe. All those transformations preserve the number of partitions so in the end the parameter is also responsible for the number of emitted files, but this is not that obvious at the point where this parameter is used. Also adding any wide transformation in between will result in different number of emitted files. So I think we can
* remove this parameter altogether - it seems that this parameter was not used before, the default value is `$UNDEFINED$` which corresponds to skipping repartition, as the result non-test execution would remain the same and test execution would be done with small amount of partitions  corresonding to test data
* keep the original parameter and introduce another parameter for the number of emitted files and set both parameters to 1 for tests
* move the repartition to the point where the datastore write happens and change the name of the parameter - this will make the name appropriate and will also set the number of partitions for pipe just as now 
* change the name of the parameter as you propose and set the parameter value to 1 for tests
* keep the implementation as it is now without name change with setting the parameter value to 1 for tests

I'd go with the first option because it removes unused parameter and does not introduce any new ones. What do you think ?

Reply:
It's good you have reminded about those transformations: remembering this I am rolling back the potential improvement of renaming this param into `documentsclassification_number_of_emitted_files`. 

The reason this parameter was introduced was to cope with the long lasting attempts. We had to increase the number of partitions in order to increase the total number of tasks doing the job. This way the scheduler can juggle the cluster resources more efficiently.

Reply:
My original idea was to completely remove the parameter but as you point out we definitely need it ;) I'm removing the parameter only from tests then.